src/components/header/EmptyThreadIndicator.js

```diff
@@ -0,0 +1,134 @@
+    shutdown?: SyntheticCssDeclarations,
+    emptyBufferStart?: SyntheticCssDeclarations,
+    emptyBufferEnd?: SyntheticCssDeclarations,
+  } {
+    const {
+      rangeStart,
+      rangeEnd,
+      width,
+      unfilteredSamplesRange,
+      interval,
+      thread: { processShutdownTime, registerTime, unregisterTime },
+    } = this.props;
+    const style = {};
+    const rangeLength = rangeEnd - rangeStart;
+    const xPixelsPerMs = width / rangeLength;
+
+    const threadEndTime =
+      unregisterTime === null ? processShutdownTime : unregisterTime;
+
+    // Did this thread startup in this time range?
+    if (registerTime > rangeStart) {
+      style.startup = {
+        left: 0,
+        width: (registerTime - rangeStart) * xPixelsPerMs,
+      };
+    }
+
+    // Did this thread shut down in this time range?
+    if (threadEndTime !== null && threadEndTime < rangeEnd) {
+      style.shutdown = {
+        right: 0,
+        width: (rangeEnd - threadEndTime) * xPixelsPerMs,
+      };
+    }
+
+    if (unfilteredSamplesRange !== null) {
+      if (
+        // Was the buffer empty at the beginning of the range?
+        unfilteredSamplesRange.start > rangeStart &&
+        // W
+        unfilteredSamplesRange.start - rangeStart > interval
+      ) {
+        const startMilliseconds = Math.max(0, registerTime - rangeStart);
+        style.emptyBufferStart = {
+          left: startMilliseconds * xPixelsPerMs,
+          right:
+            width - (unfilteredSamplesRange.start - rangeStart) * xPixelsPerMs,
+        };
+      }
+
+      if (unfilteredSamplesRange.end < rangeEnd) {
+        const endMilliseconds =
+          threadEndTime === null ? rangeEnd : Math.min(rangeEnd, threadEndTime);
+        style.emptyBufferEnd = {
+          left:
+            Math.max(0, unfilteredSamplesRange.end - rangeStart) * xPixelsPerMs,
+          width: (endMilliseconds - rangeStart) * xPixelsPerMs,
+        };
+      }
+    }
+    return style;
+  }
+
+  render() {
+    const style = this._getIndicatorPositions();
+    return (
+      <div className="headerEmptyThreadIndicator">
+        {style.startup
+          ? <DivWithTooltip
+              style={style.startup}
+              className="headerEmptyThreadIndicatorBlock headerEmptyThreadIndicatorStartup"
+              tooltip="The thread was not yet started"
+            />
+          : null}
+        {style.shutdown
+          ? <DivWithTooltip
+              style={style.shutdown}
+              className="headerEmptyThreadIndicatorBlock headerEmptyThreadIndicatorShutdown"
+              tooltip="The thread was shut down here"
+            />
+          : null}
+        {style.emptyBufferStart
+          ? <DivWithTooltip
+              style={style.emptyBufferStart}
+              className="headerEmptyThreadIndicatorBlock headerEmptyThreadIndicatorEmptyBuffer"
+              tooltip="The buffer is empty. Increase your buffer size before recording to include more samples."
```

Comment:
as discussed, I get this for this profile: http://localhost:4242/public/a926a6a13a576140b1013ab094ea2afce4fc1d47/calltree/?hiddenThreads=&react_perf&thread=4&threadOrder=0-2-3-4-5-6-7-8-1&v=2 but I don't really know why. Is it just when the main thread is too busy to get samples ?

Reply:
I'm pretty sure this is the initialization of the profiler itself. On Windows and macOS, this should be quick, because we only need to allocate the profiler buffer. On Linux, we also need to initialize lul, which is slow.

How could we stop showing misleading annotations here? I suppose we could add a "profilerInitialized" timestamp into the profile as well.

Reply:
For now I updated the copy to read:

> The buffer is empty. Either the profiler was still initializing for
> a new thread, or the profiling buffer was full. Increase your buffer
> size before recording to potentially include more samples.

Reply:
@mstange can we emit a marker for the the lul initialization? Is the initialization sync or async, e.g. are we actually missing startup samples during this time?